tests/components/onboarding/test_views.py

```diff
@@ -149,14 +160,68 @@ def auth_active(hass):
         'name': 'Test 1',
         'username': '1-user',
         'password': '1-pass',
+        'language': 'en',
     })
     resp2 = client.post('/api/onboarding/users', json={
         'client_id': CLIENT_ID,
         'name': 'Test 2',
         'username': '2-user',
         'password': '2-pass',
+        'language': 'es',
     })
 
     res1, res2 = await asyncio.gather(resp1, resp2)
 
     assert sorted([res1.status, res2.status]) == [200, 403]
+
+
+async def test_onboarding_integration(hass, hass_storage, hass_client):
+    """Test finishing integration step."""
+    mock_storage(hass_storage, {
+        'done': [const.STEP_USER]
+    })
+
+    assert await async_setup_component(hass, 'onboarding', {})
+
+    client = await hass_client()
+
+    resp = await client.post('/api/onboarding/integration', json={
+        'client_id': CLIENT_ID,
+    })
+
+    assert resp.status == 200
+    data = await resp.json()
+    assert 'auth_code' in data
+
```

Comment:
Can we add a check STEP_INTEGRATION saved in storage now

Reply:
Done